upstream-community-operators [N] [CI] universal-crossplane (1.2.1-up.3)

muvaf opened the pull request:
Opening this PR on behalf of [Upbound Inc](https://upbound.io/). [Upbound Universal Crossplane (UXP)](https://github.com/upbound/universal-crossplane) is an open-source downstream distribution of Crossplane that passes the [Crossplane Conformance](https://github.com/cncf/crossplane-conformance/) tests and it will be maintained by [Upbound Inc](https://upbound.io/).

### New Submissions

* [x] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-via-pr.md)?
* [x] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [x] Have you tested your Operator with all Custom Resource Definitions?
* [x] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [x] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md#semver-mode)?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?
* [x] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#operator-icon) set?

### Your submission should not

* [x] Modify more than one operator
* [x] Modify an Operator you don't own
* [x] Rename an operator - please remove and add with a different name instead
* [x] Submit operators to both `upstream-community-operators` and `community-operators` at once
* [x] Modify any files outside the above mentioned folders
* [x] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [ ] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://sdk.operatorframework.io/docs/overview/#operator-capability-level)

muvaf commented:
FYI, `Operator test / kiwi / Full operator test (pull_request)` seems to be stuck since it's been going on for [`3h 18m 3s`](https://github.com/operator-framework/community-operators/pull/3859/checks?check_run_id=2617217218). It took [`3m 48s`](https://github.com/operator-framework/community-operators/pull/3857/checks?check_run_id=2617218262) in the other PR, that adds the same operator.

J0zi commented:
@muvaf, please check [preview](https://operatorhub.io/preview), couple things are missing. Ignore missing channel, it is a bug in the preview. But fill other missing parts.
![image](https://user-images.githubusercontent.com/51358681/118798874-6b54df80-b89e-11eb-9ee7-1928a49385ec.png)

mvalarh commented:
/hold

mvalarh commented:
/hold cancel

mvalarh commented:
/hold

mvalarh commented:
/hold cancel

J0zi commented:
/test all

muvaf commented:
@J0zi @mvalarh is there a way to run the `Operator test / kiwi / Full operator test (pull_request)` locally? I might be help out with debugging the issue there. Seems like it gets stuck on cleanup phase.

muvaf commented:
I was able to see the logs after timing out after 6 hours.
```
TASK [deploy_olm_operator_upstream_bundle : Output the catalog-operator log to a debug file] ***
changed: [localhost]

TASK [deploy_olm_operator_upstream_bundle : Get the InstallPlan debug output] ***
changed: [localhost]

TASK [deploy_olm_operator_upstream_bundle : Get the InstallPlan debug output] ***
changed: [localhost]

TASK [deploy_olm_operator_upstream_bundle : Output the InstallPlan debug output to a debug file] ***
changed: [localhost]

TASK [deploy_olm_operator_upstream_bundle : Get namespace state] ***************
changed: [localhost]

TASK [cleanup_operator_resources : Delete all processed resources in the deploy directory] ***
changed: [localhost]
Error: The operation was canceled.
```

By looking at [the code](https://github.com/redhat-operator-ecosystem/operator-test-playbooks/blob/master/roles/cleanup_operator_resources/tasks/main.yml#L3), it tries to delete all resources in `{{ olm_operator_files_path }}`, which seems to be created & populated [here](https://github.com/redhat-operator-ecosystem/operator-test-playbooks/blob/master/roles/deploy_olm_operator_upstream_bundle/tasks/main.yml#L11). From what I can tell, it consists of `Namespace`, `OperatorGroup`, `Subscription` and `CatalogSource` which are roughly what's needed to test a full operator deployment. So I installed the operator following the guide [here](https://github.com/upbound/universal-crossplane/blob/main/cluster/olm/README.md), which is trimmed down version of [original](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) (great doc btw), and then removed each of them. After a while, the command finished and all was deleted successfully.

I tried running the [kiwi tests](https://github.com/redhat-operator-ecosystem/operator-test-playbooks/blob/upstream-community/doc/upstream/users/README.md#running-tests-from-forked-community-operators-repo-ans-specific-branch) directly on my computer, but it seems it isn't made to work with darwin-amd64, so I gave up on that.

mvalarh commented:
@muvaf i found that it hangs on

````
kubectl delete -f locks.pkg.crossplane.io.customresourcedefinition.yaml --ignore-not-found=true
customresourcedefinition.apiextensions.k8s.io "locks.pkg.crossplane.io" deleted

```

framework-automation commented:
/merge possible

muvaf commented:
@mvalarh Ah thanks! Lock is an internal object that users don't really interact with and UXP manages its own CRDs in the init container so it doesn't really need all of the CRDs to be deployed. I included them in the bundle to show up as metadata. So, I removed the locks CRD from the bundle and it seems to work now.
